src/go/wsl-helper/pkg/wsl-utils/version_windows.go

```diff
@@ -211,9 +302,62 @@ func getPackageVersion(fullName string) (*PackageVersion, error) {
 	return nil, fmt.Errorf("no info found for %s", fullName)
 }
 
+// Get the component versions for an AppX-based installation.  Returns the WSL
+// version, followed by the kernel version.
+func getAppxVersion(ctx context.Context, log *logrus.Entry) (*PackageVersion, *PackageVersion, error) {
+	newRunnerFunc := NewWSLRunner
+	if f := ctx.Value(&kWSLExeOverride); f != nil {
+		newRunnerFunc = f.(func() WSLRunner)
+	}
+	output := &bytes.Buffer{}
+	err := newRunnerFunc().WithStdout(output).WithStderr(os.Stderr).Run(ctx, "--version")
+	var exitError *exec.ExitError
+	if errors.As(err, &exitError) && exitError.ExitCode() == wslExitVersion {
+		// wsl --version is expected to return non-nil
+	} else if err != nil {
+		return nil, nil, fmt.Errorf("error running wsl --version: %w", err)
+	}
+	log.WithField("raw", output.String()).Trace("wsl --version output")
+	expr, err := regexp.Compile(`\s+\d+[.,]\d+[.,]\d+(?:[.,]\d+)?`)
```

Comment:
This looks like a duplicate of the RegExp in `UnmarshalText` above. Should it be made a package constant (with `regexp.CompileMist`)?